idaes/core/util/model_diagnostics.py

```diff
@@ -508,23 +507,31 @@ def svd_analysis(self, n_smallest_sv=10):
         
         if self.n_eq > 1:
         
+            
+            # Since we're not using svds now, no reason to accomodate its
+            # foibles
+            
             # Determine the number of singular values to compute
             # The "-1" is needed to avoid an error with svds
-            n_sv = min(n_smallest_sv, min(self.n_eq, self.n_var) - 1)
-            print("Computing the",n_sv,"smallest singular value(s)")
+            #n_sv = min(n_smallest_sv, min(self.n_eq, self.n_var) - 1)
+            #print("Computing the",n_sv,"smallest singular value(s)")
         
             # Perform SVD
             # Recall J is a n_eq x n_var matrix
             # Thus U is a n_eq x n_eq matrix
             # And V is a n_var x n_var
             # (U or V may be smaller in economy mode)
             # Thus we really only care about U
-            u, s, v = svds(self.jac_eq, k = n_sv, which='SM')
-        
+            #u, s, v = svds(self.jac_eq, k = n_sv, which='SM')#, solver='lobpcg')
+            
+            u, s, vT = svd(self.jac_eq.todense())
+            u = u[:,-1-n_smallest_sv:]
+            s = s[-1-n_smallest_sv:]
+            vT = vT[-1-n_smallest_sv:,:]
```

Comment:
I would prefer to have an option for sparse vs dense SVD, rather than hard-coding one or the other. I'd also prefer the default to be sparse to avoid the memory error that `jac_eq.todense()` will cause for large enough matrices.

Reply:
Given my experience with `svds`, I'm rather skeptical. Do we have a use case where 1) The Jacobian is too large to reasonably factor using `svd` and 2) `svds` actually converges? Maybe some of the control problems you've been looking at---it's possible their block structure works better with `svds` than a steady-state problem for a medium sized flowsheet.

Reply:
Not off the top of my head - I usually try to only call SVD on small matrices. That said, the dimension above which `todense` fails due to memory is relatively small on my (and I suspect many people's) workstation. The following code fails for me, for instance:
```python
import scipy.sparse as sps
iden = sps.identity(int(5e4))
iden = iden.todense()
```
`svds` is definitely less reliable than I'd like, but we should at least be able to try it for moderately large matrices, rather than running out of memory. I'd be fine with a try/except block that tries `svd` and runs `svds` only if it gets a memory error.